hoomd/compute.py

```diff
@@ -191,71 +86,148 @@ class thermo(_compute):
+    @log
+    def pressureYY(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureYY()
+        else:
+            return None
 
-            my_thermo.disable()
+    @log
+    def pressureYZ(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureYZ()
+        else:
+            return None
 
-        Executing the disable command will remove the thermo compute from the system. Any ```hoomd.run``` command
-        executed after disabling a thermo compute will not be able to log computed values with ``hoomd.analyze.log``.
+    @log
+    def pressureZZ(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPressureZZ()
+        else:
+            return None
 
-        A disabled thermo compute can be re-enabled with :py:meth:`enable()`.
-        """
+    @log
+    def kineticEnergy(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getKineticEnergy()
+        else:
+            return None
 
-        _compute.disable(self)
+    @log
+    def translationalKineticEnergy(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getTranslationalKineticEnergy()
+        else:
+            return None
 
-        hoomd.context.current.thermos.remove(self)
+    @log
+    def rotationalKineticEnergy(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getRotationalKineticEnergy()
+        else:
+            return None
 
-    def enable(self):
-        R""" Enables the thermo compute.
+    @log
+    def potentialEnergy(self):
+        if self._attached:
+            self._cpp_obj.compute(self._simulation.timestep)
+            return self._cpp_obj.getPotentialEnergy()
+        else:
+            return None
 
-        Examples::
+    @log
+    def NDOF(self):
+        if self._attached:
+            return self._cpp_obj.getNDOF()
+        else:
+            return None
 
-            my_thermo.enable()
+    @log
+    def translationalDOF(self):
+        if self._attached:
+            return self._cpp_obj.getTranslationalDOF()
+        else:
+            return None
 
-        See ``disable``.
-        """
+    @log
+    def rotationalDOF(self):
+        if self._attached:
+            return self._cpp_obj.getRotationalDOF()
+        else:
+            return None
 
-        _compute.enable(self)
+    @log
+    def numParticles(self):
+        if self._attached:
+            return self._cpp_obj.getNumParticles()
+        else:
+            return None
 
-        hoomd.context.current.thermo.append(self)
 
-class thermoHMA(_compute):
+class thermoHMA(_Compute):
```

Comment:
If we are going to change the parent class for this class then we can just remove the `_compute` class.

Reply:
I left this because there are classes in hpmc that still have `_compute` as their base class, but I can change those too.